FileExists panics w/o read permission

See https://github.com/romie-gr/romie/blob/f4f9b85971f907970588030f8344ce86542495bd/internal/utils/filesystem.go#L16-L21

there is no check for permissions, as a result the value of `info` variable is `nil`.
As a result, if you create a testcase where the path has not read permissions, then the final return statement https://github.com/romie-gr/romie/blob/f4f9b85971f907970588030f8344ce86542495bd/internal/utils/filesystem.go#L23 panics, because it's doing: `true == nil` or `false == nil`.

I added a testcase to verify this:

```go
    --- FAIL: TestFileExists/Returns_false_when_file_is_into_a_folder_without_read_permissions (0.00s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x111e1be]

goroutine 22 [running]:
testing.tRunner.func1.1(0x113b2a0, 0x1248820)
        /usr/local/go/src/testing/testing.go:1076 +0x30d
testing.tRunner.func1(0xc000082f00)
        /usr/local/go/src/testing/testing.go:1079 +0x41a
panic(0x113b2a0, 0x1248820)
        /usr/local/go/src/runtime/panic.go:969 +0x175
github.com/romie-gr/romie/internal/utils.exists(0xc00009e090, 0x2a, 0x100000000, 0xc00004a748)
        /Users/drpaneas/gocode/src/github.com/romie-gr/romie/internal/utils/filesystem.go:23 +0x5e
github.com/romie-gr/romie/internal/utils.FileExists(...)
        /Users/drpaneas/gocode/src/github.com/romie-gr/romie/internal/utils/filesystem.go:33
github.com/romie-gr/romie/internal/utils.TestFileExists.func1(0xc000082f00)
        /Users/drpaneas/gocode/src/github.com/romie-gr/romie/internal/utils/filesystem_test.go:118 +0x45
testing.tRunner(0xc000082f00, 0xc000092510)
        /usr/local/go/src/testing/testing.go:1127 +0xef
created by testing.(*T).Run
        /usr/local/go/src/testing/testing.go:1178 +0x386
exit status 2
FAIL    github.com/romie-gr/romie/internal/utils        0.300s
```

I found this issue while I was working on the `CreateFile` card.

--- a/internal/utils/filesystem.go
+++ b/internal/utils/filesystem.go
@@ -15,7 +15,7 @@ func exists(path string, isDir bool) bool {
 
 	info, err := os.Stat(path)
 	if err != nil {
-		if os.IsNotExist(err) {
+		if os.IsNotExist(err) || os.IsPermission(err) {
 			return false
 		}
 	}

--- a/internal/utils/filesystem_test.go
+++ b/internal/utils/filesystem_test.go
@@ -77,6 +77,7 @@ func TestFileExists(t *testing.T) {
 	if err := os.Mkdir(nonWritableDir, 0400); err != nil {
 		log.Fatalf("Cannot create non writable directory %q", nonWritableDir)
 	}
+
 	tests := []struct {
 		name string
 		path string
@@ -113,6 +114,7 @@ func TestFileExists(t *testing.T) {
 			false,
 		},
 	}
+
 	for _, tt := range tests {
 		tt := tt
 		t.Run(tt.name, func(t *testing.T) {
@@ -121,5 +123,6 @@ func TestFileExists(t *testing.T) {
 			}
 		})
 	}
+
 	_ = os.RemoveAll(nonWritableDir)
 }

--- a/internal/utils/filesystem_test.go
+++ b/internal/utils/filesystem_test.go
@@ -94,11 +94,6 @@ func TestFileExists(t *testing.T) {
 			false,
 		},
 		{
-			"Returns false when given file and its parent folder do not exist",
-			filepath.Join(nonExistingFolder, "missing-file.txt"),
-			false,
-		},
-		{
 			"Returns false when file is into a folder without read permissions",
 			filepath.Join(nonWritableDir, "missing-file.txt"),
 			false,

--- a/internal/utils/filesystem_test.go
+++ b/internal/utils/filesystem_test.go
@@ -13,6 +13,7 @@ var (
 	nonExistingFolder = "./testdata/a-folder-that-does-not-exist"
 	existingFile      = "./testdata/a-folder-that-exists/file.txt"
 	nonExistingFile   = "./testdata/a-folder-that-exists/missing-file.txt"
+	nonWritableDir    = "./testdata/non-writable-dir"
 )
 
 func ExampleFolderExists() {

--- a/internal/utils/filesystem_test.go
+++ b/internal/utils/filesystem_test.go
@@ -2,6 +2,9 @@ package utils
 
 import (
 	"fmt"
+	"log"
+	"os"
+	"path/filepath"
 	"testing"
 )
 
@@ -70,6 +73,9 @@ func ExampleFileExists() {
 }
 
 func TestFileExists(t *testing.T) {
+	if err := os.Mkdir(nonWritableDir, 0400); err != nil {
+		log.Fatalf("Cannot create non writable directory %q", nonWritableDir)
+	}
 	tests := []struct {
 		name string
 		path string
@@ -83,6 +89,16 @@ func TestFileExists(t *testing.T) {
 		{
 			"Returns false when given file does not exist",
 			nonExistingFile,
+			false,
+		},
+		{
+			"Returns false when given file and its parent folder do not exist",
+			filepath.Join(nonExistingFolder, "missing-file.txt"),
+			false,
+		},
+		{
+			"Returns false when file is into a folder without read permissions",
+			filepath.Join(nonWritableDir, "missing-file.txt"),
 			false,
 		},
 		{
@@ -104,4 +120,5 @@ func TestFileExists(t *testing.T) {
 			}
 		})
 	}
+	_ = os.RemoveAll(nonWritableDir)
 }

